src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -128,40 +149,76 @@ private void EnsureWorkerIsRunning()
                         var thisRef = (MultiAgent)s;
                         try
                         {
-                            // Do the actual processing
-                            thisRef.EventSourceTrace("MultiAgent worker running");
-                            thisRef.WorkerLoop();
+                            try
+                            {
+                                // Do the actual processing
+                                thisRef.EventSourceTrace("MultiAgent worker running");
+                                thisRef.WorkerLoop();
+                            }
+                            finally
+                            {
+                                thisRef.EventSourceTrace("MultiAgent worker shutting down");
+
+                                // The multi handle's reference count was increased prior to launching
+                                // this processing task.  Release that reference; any Dispose operations
+                                // that occurred during the worker's processing will now be allowed to
+                                // proceed to clean up the multi handle.
+                                thisRef._multiHandle.DangerousRelease();
```

Comment:
Maybe this task method should be a distinct method... it took a fair bit of brainpower to notice that this wasn't actually backwards (since the call to AddRef is after several }s, but before the task start.

Or maybe I'm just not brain-trained in Task-fu